### PR TITLE
FFC-154605: Send session events to populate auto eligibility MI report

### PR DESCRIPTION
**Pull request by marcin-mogiela**
https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/154605

### CODE REVIEW

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js [hunk 1]
```diff
@@ -10,9 +12,55 @@ const processEligibleSbi = async (customer) => {
...
+          message: 'The customer has been put on the waiting list',
+          data: {
+            customer: {
+              sbi: customer.sbi,
+              crn: customer.crn,
+              businessEmail: customer.businessEmail
+            }
+          },
+          raisedOn: new Date(),
+          raisedBy: 'auto-eligibility:incoming-register-your-interest'
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/register-your-interest/process-eligible-sbi.js:34]: looking at the other MI report the raised by is set to the business email - probably best to match that even if in our case this is a system driven action

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js [hunk 2]
```diff
@@ -10,9 +12,55 @@ const processEligibleSbi = async (customer) => {
...
+        }
+      }
+    })
   } else {
     if (customer.businessEmailHasMultipleDistinctSbi) {
       console.log(`${new Date().toISOString()} The customer's business email has multiple distinct SBI`)
+      await raiseEvent({
+        name: 'auto-eligibility:incoming-register-your-interest:rejected_due_to_multiple_sbi',
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/register-your-interest/process-eligible-sbi.js:42]: maybe we can make an enum/constant with all the event names?

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js [hunk 3]
```diff
@@ -21,6 +69,29 @@ const processEligibleSbi = async (customer) => {
     }
     await customerDbTable.updateWaitingUpdatedAt(customer.sbi, customer.crn)
     await emailNotifier.sendWaitingListEmail(customer.businessEmail)
+    await raiseEvent({
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/register-your-interest/process-eligible-sbi.js:72]: Possible duplicate of app/auto-eligibility/register-your-interest/process-eligible-sbi.js line 15

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js (hunk 1 above)
Comment (marcin-mogiela) [reply app/auto-eligibility/register-your-interest/process-eligible-sbi.js]: done

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js (hunk 2 above)
Comment (marcin-mogiela) [reply app/auto-eligibility/register-your-interest/process-eligible-sbi.js]: done

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js (hunk 3 above)
Comment (marcin-mogiela) [reply app/auto-eligibility/register-your-interest/process-eligible-sbi.js]: true but this duplication is due to feature toggle and will be removed once feature toggle is removed

now it's for simplicity

**Comment by liamwilson-kainos** [comment]
image

await eventPublisher.sendEvent(event)

File: app/auto-eligibility/telemetry/raise-telemetry-event.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const raiseEvent = require('../../event/raise-event')
+const appInsightsConfig = require('../../app-insights/app-insights.config')
+
+module.exports = (customer) => {
+  return async (type, message, data = customer) => {
+    await raiseEvent({
+      name: 'send-session-event',
+      properties: {
+        id: `${customer.sbi}_${customer.crn}`,
+        sbi: customer.sbi,
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/telemetry/raise-telemetry-event.js:10]: Needs to be a string https://github.com/DEFRA/ffc-ahwr-event/blob/6551b6056b78018857f99f01b66dacdf57027a48/ffc-ahwr-event/event-schema.js#L7

**Review by liamwilson-kainos — CHANGES_REQUESTED** [review]

**Review by liamwilson-kainos — APPROVED** [review]

File: app/auto-eligibility/register-your-interest/process-eligible-sbi.js [hunk 5]
```diff
@@ -21,6 +41,15 @@ const processEligibleSbi = async (customer) => {
     }
     await customerDbTable.updateWaitingUpdatedAt(customer.sbi, customer.crn)
     await emailNotifier.sendWaitingListEmail(customer.businessEmail)
+    await raiseTelemetryEvent(customer)(
+      telemetryEvent.PUT_ON_THE_WAITING_LIT,
```
Comment (json-hamilton) [inline app/auto-eligibility/register-your-interest/process-eligible-sbi.js:45]: here also

File: app/auto-eligibility/telemetry/raise-telemetry-event.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+const config = require('../../config')
+const appInsightsConfig = require('../../app-insights/app-insights.config')
+
+module.exports = (customer) => {
+  const eventPublisher = new PublishEvent(config.mqConfig.eventQueue)
+  return async (type, message, data) => {
+    await eventPublisher.sendEvent({
+      name: 'send-session-event',
+      properties: {
+        id: `${customer.sbi}_${customer.crn}`,
```
Comment (liamwilson-kainos) [inline app/auto-eligibility/telemetry/raise-telemetry-event.js:11]: properties.status is missing and is required i believe (fails locally for me)

**Review by liamwilson-kainos — APPROVED** [review]

**Pull request merged**